### PR TITLE
✨ Update CI to use OpenStack Bobcat

### PR DESCRIPTION
**Pull request by EmilienM**
**What this PR does / why we need it**:

Update devstack configuration to deploy OpenStack Bobcat with OVN on Ubuntu 22.04.

### CODE REVIEW

**Comment by EmilienM** [comment]
I know Antelope is the latest stable version but I want to see how it works with Zed first (step by step).

File: hack/ci/cloud-init/controller.yaml.tpl [hunk 1]
```diff
@@ -35,9 +35,12 @@
     ENABLED_SERVICES+=,g-api
 
     # Neutron
-    ENABLED_SERVICES+=,neutron-api,neutron-agent,neutron-dhcp,neutron-l3,neutron-trunk
+    ENABLED_SERVICES+=,q-svc,q-trunk,ovn-controller,ovs-vswitchd,ovn-northd,ovsdb-server,q-ovn-metadata-agent
+    Q_AGENT=ovn
```
Comment (MaysaMacedo) [inline hack/ci/cloud-init/controller.yaml.tpl:39]: ovn is already the [default agent on zed](https://github.com/openstack/devstack/blob/stable/zed/lib/neutron_plugins/ml2#L11), we likely don't need to directly set this agent. Same for the `Q_ML2_PLUGIN_MECHANISM_DRIVERS` and `Q_ML2_TENANT_NETWORK_TYPE`

File: hack/ci/cloud-init/worker.yaml.tpl [hunk 2]
```diff
@@ -27,6 +27,11 @@
 
     # Neutron
     enable_plugin neutron https://github.com/openstack/neutron stable/${OPENSTACK_RELEASE}
+    ENABLED_SERVICES+=,ovn-controller,ovs-switchd,ovsdb-server,q-fake,q-ovn-metadata-agent
+    Q_AGENT=ovn
```
Comment (MaysaMacedo) [inline hack/ci/cloud-init/worker.yaml.tpl:31]: ditto

File: docs/book/src/development/ci.md [hunk 3]
```diff
@@ -66,6 +66,8 @@ In addition, DevStack will create a floating IP network using CIDR defined in `F
 
 The effect of this is that instances created on either `controller` or `worker` can get a floating ip from the `public` network. Traffic using this floating IP will be routed via `controller` and externally via NAT.
 
+We are configuring OVN to provide default DNS servers if a subnet if created without specifying DNS servers. This can be overridden in `OPENSTACK_DNS_NAMESERVERS`.
```
Comment (MaysaMacedo) [inline docs/book/src/development/ci.md:69]: typo:
if a subnet **is** created

**Comment by EmilienM** [comment]
We have made progress with DNS, now we have clusters failing to deploy because the subnode seems not ready yet:

```
E0110 19:44:18.816900       1 controller.go:329] "Reconciler error" err="create OpenStack instance: error creating Openstack instance: Bad request with: [POST http://10.0.3.15/compute/v2.1/servers], error message: {\"badRequest\": {\"code\": 400, \"message\": \"The requested availability zone is not available\"}}" controller="openstackmachine" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackMachine" OpenStackMachine="e2e-quie8x/cluster-e2e-quie8x-md-invalid-az-ppczg" namespace="e2e-quie8x" name="cluster-e2e-quie8x-md-invalid-az-ppczg" reconcileID="a4755526-793d-4db7-810d-af15e62f561e"
```

Source:
* https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1804/pull-cluster-api-provider-openstack-e2e-test/1745154653198225408/artifacts/clusters/bootstrap/logs/capo-system/capo-controller-manager/capo-controller-manager-66896777f9-jbnhh/manager.log
* https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1804/pull-cluster-api-provider-openstack-e2e-test/1745154653198225408/artifacts/clusters/bootstrap/e2e-quie8x/machines/cluster-e2e-quie8x-control-plane-qcn5z/

**Comment by EmilienM** [comment]
Another cluster failed and it's suspicious: https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1804/pull-cluster-api-provider-openstack-e2e-test/1745154653198225408/artifacts/clusters/bootstrap/e2e-z5qi8n/machines/cluster-e2e-z5qi8n-control-plane-t99mk/
I haven't found why yet.

**Comment by mandre** [comment]
On the last test run, the `cluster-e2e-6qwkmm-control-plane-7q5kv` vm fails to boot. From the console logs I see:
```
[    8.326287] audit: type=1130 audit(1704964471.402:41): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=ignition-files comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
```

**Comment by EmilienM** [comment]
We're removing the bastion in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1796/commits/72146dc38b5756f3a391f0e30c5f1a98844de218 - maybe that will help

**Comment by EmilienM** [comment]
/test pull-cluster-api-provider-openstack-e2e-test
in case I'm lucky this time

**Comment by EmilienM** [comment]
/hold
I cherry-picked a commit that removes the bastions in e2e, I want to see if that helps with instances going into ERROR state.

**Comment by EmilienM** [comment]
it's green! let's wait for https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1796 to merge.

File: test/e2e/data/kustomize/common-patches/cni/patch-cluster.yaml [hunk 4]
```diff
@@ -1,12 +1,4 @@
 ---
-- op: add
-  path: /spec/bastion
-  value:
-    enabled: true
-    instance:
-      flavor: ${OPENSTACK_BASTION_MACHINE_FLAVOR}
-      image: ${OPENSTACK_BASTION_IMAGE_NAME}
-      sshKeyName: ${OPENSTACK_SSH_KEY_NAME}
```
Comment (mdbooth) [inline test/e2e/data/kustomize/common-patches/cni/patch-cluster.yaml:9]: I think we've come full circle on this: we need to keep the Bastions for logging.

**Comment by EmilienM** [comment]
The tests work fine when we delete the bastion nodes, which make me think switching to bobcat highlight some resources issues in our jobs where some nodes goes into ERROR because "no valid host found" in Nova.
Maybe we need to reduce the number of concurrent clusters, or add MachineHealthChecks so Machines get re-created if an error, or mark heavy tests and avoid concurrency for these ones.

**Comment by EmilienM** [comment]
when we reduce the control plane flavor, it seems to be fine.

@lentzi90 @jichenjc @mdbooth wdyt?

**Comment by jichenjc** [comment]
>>>when we reduce the control plane flavor, it seems to be fine.

I don't remember whether we have anything has impact on the performance, so seems your change (vcpu 4=>2) should be good

**Comment by lentzi90** [comment]
2 vcpu should be fine for what we are doing, it is not like we stress the control plane with anything.
Let's check the full test also
/test pull-cluster-api-provider-openstack-e2e-full-test

**Comment by jichenjc** [comment]
@lentzi90 seems the CI not finished but it's merged, is the CI e2e test not a gate anymore (need pass in order to merge) 

image

**Comment by lentzi90** [comment]
Oh! The e2e test is a gate but not the e2e-full. I should have realized to put hold. Well I guess we will soon know if it works or not :slightly_smiling_face:

**Comment by EmilienM** [comment]
Thanks for merging. Yes I'll keep an eye on that job, I run it nearly every day these days... :)

**Pull request merged**